### DISCUSSION TITLE
Using `auto-py-to-exe` for a python package with relative import

### DISCUSSION THREAD

**Discussion by LucBerge** [brentvollebregt/auto-py-to-exe#336] [General]
Hey ! I want to share with you a workaround to build an `.exe` file for a python package using relative imports.

I have a project called [Wallme on Github](https://github.com/LucBerge/wallme) which is a pip package. I want to create an `.exe` file with `auto-py-to-exe` to allow non-dev users to use it. As script location, I used my `main.py` which is the entry point of my package. The `.exe` is built but when I run it, I get the following error:
```
ImportError: attempted relative import with no known parent package
```
This is not an `auto-py-to-exe` error, it is `pyinstaller` related. [I found this issue to bypass the error](https://github.com/pyinstaller/pyinstaller/issues/2560).

> The work around is to create a small "stub" application that lives outside of the package and calls into it:
> ```python
> from foo import main
> main()
> ```
> ...and then point PyInstaller at the stub.

I can tell you it's working for me.

**Comment by brentvollebregt** [comment]
Thanks for sharing this! I do this from time to time to package modules/libraries 🏄 

I've got a decent example of [doing it here](https://github.com/brentvollebregt/whos-on-my-network/blob/master/run.py).
